flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/factory/KubernetesJobManagerFactory.java

```diff
@@ -106,6 +106,7 @@ private static Deployment createJobManagerDeployment(
                 .withName(
                         KubernetesUtils.getDeploymentName(
                                 kubernetesJobManagerParameters.getClusterId()))
+                .addToAnnotations(kubernetesJobManagerParameters.getAnnotations())
```

Comment:
Could we use `withAnnotations` here? AFAIK, we should only set the `kubernetesJobManagerParameters.getAnnotations()` to the deployment.

Reply:
Done.